application/src/main/java/org/togetherjava/tjbot/features/moderation/TransferQuestionCommand.java

```diff
@@ -0,0 +1,182 @@
+        event.getJDA()
+            .retrieveUserById(args.get(0))
+            .map(user -> user)
+            .flatMap(fetchedUser -> createForumPost(event, fetchedUser))
+            .flatMap(user -> dmUser(event.getChannel(), user))
+            .flatMap(helperForum -> event.reply("transferred to" + helperForum.getAsMention())
+                .setEphemeral(true))
+            .flatMap(info -> deleteOriginalMessage(event.getJDA(), args.get(2), args.get(1)))
+            .queue();
+    }
+
+    private static String createTitle(String message) {
+        if (message.length() >= TITLE_MAX_LENGTH) {
+            int lastWordEnd = message.lastIndexOf(' ', TITLE_MAX_LENGTH);
+
+            if (lastWordEnd == -1) {
+                lastWordEnd = TITLE_MAX_LENGTH;
+            }
+
+            message = message.substring(0, lastWordEnd);
+        }
+
+        return isTitleValid(message) ? message : "Untitled";
+    }
+
+    private static boolean isTitleValid(CharSequence title) {
+        String titleCompact = TITLE_COMPACT_REMOVAL_PATTERN.matcher(title).replaceAll("");
+
+        return titleCompact.length() >= TITLE_COMPACT_LENGTH_MIN
+                && titleCompact.length() <= TITLE_COMPACT_LENGTH_MAX;
+    }
+
+    private RestAction<User> createForumPost(ModalInteractionEvent event, User originalUser) {
+        MessageCreateData forumMessage = MessageCreateData
+            .fromContent(event.getValue(TRANSFER_QUESTION_INPUT_ID).getAsString());
+        String forumTitle = event.getValue(TRANSFER_QUESTION_TITLE_ID).getAsString();
+        ForumChannel questionsForum = getHelperForum(event.getJDA());
+        List<ForumTag> tags = questionsForum.getAvailableTagsByName("JAVA", true);
+        ForumTag javaTag = tags.get(0);
+
+        return questionsForum.createForumPost(forumTitle, forumMessage)
+            .setTags(ForumTagSnowflake.fromId(javaTag.getId()))
+            .map(ForumPost::getMessage)
+            .flatMap(message -> message.reply("Original Post from " + originalUser.getAsMention()))
+            .map(sent -> originalUser);
+
+    }
+
+    private RestAction<ForumChannel> dmUser(MessageChannelUnion sourceChannel, User originalUser) {
+
+        return originalUser.openPrivateChannel()
+            .flatMap(channel -> channel.sendMessage(
+                    "Hi, TJ here. You are getting this message " + "because you tried to ask a "
+                            + "question in wrong chat, please read rules - Together Java"))
+            .onErrorFlatMap(error -> sourceChannel
+                .sendMessage("Due to failed dm interaction, you are getting "
+                        + "this message here. Please read rules about asking questions - Together Java"
+                        + originalUser.getAsMention()))
+            .onSuccess(failed -> logger.warn("failed to dm user {} with id: {}",
+                    originalUser.getName(), originalUser.getId()))
```

Comment:
what is this

Reply:
Didn't realise, onSuccess triggered right after the chain it follows. It's removed.